src/mlpack/methods/reinforcement_learning/replay/prioritized_replay.hpp

```diff
@@ -118,6 +160,37 @@ class PrioritizedReplay
     }
   }
 
+  /**
+   * Get the reward, next state and terminal boolean for nth step.
+   *
+   * @param reward Given reward.
+   * @param nextState Given next state.
+   * @param isEnd Whether next state is terminal state.
+   * @param discount The discount parameter.
+   */
+  void GetNStepInfo(double& reward,
+                    StateType& nextState,
+                    bool& isEnd,
+                    const double& discount)
+  {
+    reward = nStepBuffer.back().reward;
+    nextState = nStepBuffer.back().nextState;
+    isEnd = nStepBuffer.back().isEnd;
+
+    // Should start from the second last transition in buffer.
+    for (int i = nStepBuffer.size() - 2; i >= 0; i--)
+    {
+      double r = nStepBuffer[i].reward;
+      bool iE = nStepBuffer[i].isEnd;
+      reward = r + discount * reward * (1 - iE);
+      if (iE)
+      {
+        nextState = nStepBuffer[i].nextState;
+        isEnd = iE;
```

Comment:
Couldn't we break here?

Reply:
For the case when the buffer contains more than one terminals (i.e. value of `isEnd` being `1`), we'd want to take the `isEnd` occuring first from begining of buffer(or last from the end of the buffer). I hope I made sense.

Reply:
You are right, thanks for the clarification.